src/main/java/seedu/address/model/lesson/Time.java

```diff
@@ -0,0 +1,135 @@
+
+    public static final double MINIMUM_DURATION = 0.5;
+
+    public static final String MESSAGE_CONSTRAINTS_INVALID_DAY =
+            "Day of week should be an integer in the range [1, 7]";
+
+    public static final String MESSAGE_CONSTRAINTS_INVALID_LOCALTIME =
+            "The start or end time should be in HH:MM format";
+
+    public static final String MESSAGE_CONSTRAINTS_IMPROPER_TIME =
+            "Start time should be before end time";
+
+    public static final String MESSAGE_CONSTRAINTS_INVALID_DURATION =
+            "Lesson duration must be at least " + MINIMUM_DURATION + " hours";
+
+    private DayOfWeek dayOfOccurrence;
+    @JsonFormat(pattern = "HH:mm")
+    private LocalTime startTime;
+    @JsonFormat(pattern = "HH:mm")
+    private LocalTime endTime;
+
+    private double duration;
+
+    /**
+     * Constructs a {@code Time}.
+     *
+     * @param dayOfOccurrence Which day the lesson occurs.
+     * @param startTime The time the lesson starts.
+     * @param endTime The time the lesson ends.
+     */
+    public Time(DayOfWeek dayOfOccurrence, LocalTime startTime, LocalTime endTime) {
+        requireAllNonNull(dayOfOccurrence, startTime, endTime);
+        checkArgument(startTime.isBefore(endTime), MESSAGE_CONSTRAINTS_IMPROPER_TIME);
+
+        duration = computeDuration(startTime, endTime);
+        checkArgument(isValidDuration(duration, MINIMUM_DURATION), MESSAGE_CONSTRAINTS_INVALID_DURATION);
+
+        this.dayOfOccurrence = dayOfOccurrence;
+        this.startTime = startTime;
+        this.endTime = endTime;
+    }
+
+    /**
+     * Default constructor to aid Jackson in deserializing the class.
+     * Solution adapted from https://www.baeldung.com/jackson-exception#2-the-solution
+     */
+    public Time() {
+    }
+
+    /**
+     * Computes the number of hours between start time and end time.
+     *
+     * @param startTime The start time.
+     * @param endTime The end time.
+     * @return A double representing the number of hours.
+     */
+    private double computeDuration(LocalTime startTime, LocalTime endTime) {
+        long durationInMinutes = Duration.between(startTime, endTime).toMinutes();
+        return durationInMinutes / 60.0;
+    }
+
+    /**
+     * Returns true if the time interval between the given start time and the given end time
+     * is at least the minimum duration.
+     */
+    public static boolean isValidDuration(double duration, double minDuration) {
+        return duration >= minDuration;
+    }
+
+    public DayOfWeek getDayOfOccurrence() {
+        return dayOfOccurrence;
+    }
+
+    public LocalTime getStartTime() {
+        return startTime;
+    }
+
+    public LocalTime getEndTime() {
+        return endTime;
+    }
+
+    public double getDuration() {
+        return duration;
+    }
+
+    /**
+     * Returns true if both times have the same day of occurrence, start time and end time.
+     */
+    @Override
+    public boolean equals(Object other) {
```

Comment:
I think this is the portion of the code (as discussed just now) which has to be edited to include the algorithm for de-conflicting lesson times?

Reply:
Nice, I just saw the tests and they appear to be very comprehensive in testing different overlap variations.